Objective-C/CBLDatabase.mm

```diff
@@ -192,6 +194,8 @@ - (CBLDocument*) documentWithID: (NSString*)documentID {
 
 
 - (CBLDocumentFragment*) objectForKeyedSubscript: (NSString*)documentID {
+    CBLAssertNotNil(documentID);
```

Comment:
This check isn't necessary, since the `documentID` is just passed to `[self documentWithID:]`, which checks.